services/workflows-service/src/data-migration/scripts/sync/sync.intg.test.ts

```diff
@@ -34,6 +32,202 @@ describe('Data Sync System:', () => {
     await cleanupDatabase();
     await tearDownDatabase();
   });
+  describe('mergeSyncObjects', () => {
+    it('should merge objects with shared dry run environments', () => {
+      const objects: SyncedObject[] = [
```

Comment:
Just a nice to know - many places that you'd add a type like this using the `satisfies` keyword instead would be preferred, including here and especially for the `columns` property.